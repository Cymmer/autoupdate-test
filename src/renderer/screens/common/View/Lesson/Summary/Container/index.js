import React from 'react';
import PropTypes from 'prop-types';
import { Switch, Route, Redirect } from 'react-router-dom';
import GLOBALS from 'codechum-app-globals';

import { Container, Header, Text } from 'components/elements';
import { tabKinds, textTypes } from 'components/elements/constants';

import { LessonImage } from 'components';

import python from 'static/images/Icons/python.png';
import styles from './styles.module.scss';

import summaryTabs from './constants/summaryTabs';

const SummaryContainer = ({
  lesson,
  baseRoute,
  baseLink,
  glossarySyntaxList,
  glossaryTermsList,
  activeTab,
}) => {
  const tabs = [];
  if (glossarySyntaxList) {
    tabs.push({
      name: summaryTabs.SYNTAX.name,
      value: summaryTabs.SYNTAX.value,
      kind: tabKinds.LINK,
      action: `${baseLink}/${summaryTabs.SYNTAX.value}`,
    });
  }
  if (glossaryTermsList) {
    tabs.push({
      name: summaryTabs.TERMS.name,
      value: summaryTabs.TERMS.value,
      kind: tabKinds.LINK,
      action: `${baseLink}/${summaryTabs.TERMS.value}`,
    });
  }

  const link = glossarySyntaxList
    ? `${baseLink}/${summaryTabs.SYNTAX.value}`
    : `${baseLink}/${summaryTabs.TERMS.value}`;

  return (
    <>
      <Header activeTab={activeTab} tabs={tabs}>
        <div className={styles.Summary_title}>
          <LessonImage
            className={styles.Summary_title_image}
            colorName={GLOBALS.COLOR_NAMES.BLUE}
            image={python}
          />
          <div>
            <Text
              colorClass={GLOBALS.COLOR_CLASSES.NEUTRAL['400']}
              type={textTypes.BODY.MD}
            >
              Lesson Summary
            </Text>
            <Text type={textTypes.HEADING.MD}>{lesson.title}</Text>
          </div>
        </div>
      </Header>

      <Container className={styles.Summary_container}>
        <Switch>
          <Route
            name="Summary - Syntax"
            path={`${baseRoute}/${summaryTabs.SYNTAX.value}`}
            render={() => glossarySyntaxList}
          />
          <Route
            name="Summary - Terms"
            path={`${baseRoute}/${summaryTabs.TERMS.value}`}
            render={() => glossaryTermsList}
          />
          <Redirect to={link} />
        </Switch>
      </Container>
    </>
  );
};

SummaryContainer.propTypes = {
  lesson: PropTypes.object.isRequired,
  baseRoute: PropTypes.string.isRequired,
  baseLink: PropTypes.string.isRequired,

  activeTab: PropTypes.oneOf([
    summaryTabs.SYNTAX.value,
    summaryTabs.TERMS.value,
  ]).isRequired,

  // a JSX component containing the glossary syntax list
  glossarySyntaxList: PropTypes.node.isRequired,

  // a JSX component containing the glossary terms list
  glossaryTermsList: PropTypes.node.isRequired,
};

export default SummaryContainer;
