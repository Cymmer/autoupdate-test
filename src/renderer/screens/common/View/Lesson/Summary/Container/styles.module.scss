.Summary {
  &_title {
    display: flex;
    align-items: center;

    &_image {
      height: 88px;
      width: 88px;
      background-size: 56px;
      border-radius: 16px;
      margin: 0 32px 0 0;
    }
  }

  &_container {
    padding: 24px 1rem 120px 1rem;
  }
}
