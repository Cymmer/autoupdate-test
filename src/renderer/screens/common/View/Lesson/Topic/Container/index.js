import React from 'react';
import PropTypes from 'prop-types';
import GLOBALS from 'codechum-app-globals';
import styles from './styles.module.scss';

import { Cody, Text, Container } from '../../../../../../components/elements';
import { textTypes } from '../../../../../../components/elements/constants';

const TopicContainer = ({ number, title, titleJsx, image, children }) => (
  <Container className={styles.TopicContainer}>
    <Text
      className={styles.TopicContainer_topic}
      colorClass={GLOBALS.COLOR_CLASSES.NEUTRAL['300']}
      type={textTypes.HEADING.XXS}
    >
      Topic {number}
    </Text>

    <div className={styles.TopicContainer_title}>
      <Cody className={styles.TopicContainer_title_cody} src={image} />
      <div className={styles.TopicContainer_title_content}>
        <Text
          className={styles.TopicContainer_title_text}
          type={textTypes.HEADING.MD}
        >
          {title}
        </Text>
        {titleJsx}
      </div>
    </div>

    {children}
  </Container>
);

TopicContainer.propTypes = {
  number: PropTypes.number.isRequired,
  title: PropTypes.string.isRequired,
  image: PropTypes.any.isRequired,
  titleJsx: PropTypes.node.isRequired,
  children: PropTypes.node.isRequired,
};

export default TopicContainer;
