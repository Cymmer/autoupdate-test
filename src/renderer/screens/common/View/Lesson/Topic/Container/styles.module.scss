@import '../../../../../../scss/variables';
@import '../../../../../../scss/mixins';
@import '../../../../../../scss/themes';

.TopicContainer {
  padding: 56px 1rem;

  @include breakpoint(md) {
    padding: 56px 1rem 96px;
  }

  &_topic {
    margin: 0 0 48px 0;
  }

  &_title {
    display: flex;

    @include breakpoint(md) {
      flex-direction: column;
    }

    &_cody {
      height: 100%;
      width: 100%;
      max-width: 240px;
      margin: 0 48px 24px 0;

      @include breakpoint(sm) {
        max-width: 160px;
      }
    }

    &_content {
      width: calc(100% - 288px);

      @include breakpoint(md) {
        width: 100%;
      }

      > p {
        margin: 0 0 16px 0;

        > code {
          color: $red-300;
        }
      }

      > div > pre {
        margin: 16px 0 !important;
      }
    }

    &_text {
      margin: 0 0 16px 0;
    }
  }
}
