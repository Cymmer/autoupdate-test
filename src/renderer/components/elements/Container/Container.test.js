import React from 'react';
import { shallow } from 'enzyme';
import Container from './index';

describe('RENDERING - Container', () => {
  it('should render with default props', () => {
    const component = shallow(<Container />);
    expect(component).toMatchSnapshot();
  });

  it('should render its children', () => {
    const component = shallow(<Container>Test</Container>);
    expect(component).toMatchSnapshot();
  });
});
