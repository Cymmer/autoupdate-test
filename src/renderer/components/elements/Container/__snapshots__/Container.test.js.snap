// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RENDERING - Container should render its children 1`] = `
<div
  className="Container"
>
  Test
</div>
`;

exports[`RENDERING - Container should render with default props 1`] = `
<div
  className="Container"
/>
`;
